### DISCUSSION TITLE
Hidden window stays hidden, missing an event

### DISCUSSION THREAD

**Discussion by keyle** [wailsapp/wails#1541] [Q&A]
Hi,

On MacOS, I've got my application window hiding on `cmd+h`. It works as expected. However when I click the app in the dock, I'd expect it to show again. 

Is there an event I should listen to to call `WindowShow`? 

Thanks

**Comment by leaanthony** [comment]
Hide is not equivalent to minimise, which sounds like what you want: https://wails.io/docs/reference/runtime/window#windowminimise

**Reply by keyle** [reply]
Once again, thanks for the prompt response.

At first glance it appears to be the solution, although macos will show the minimize animation instead of just hiding the window like other software does. That's not a big deal, however... alt-tabbing (cmd+tab) back to application should show the application again, if it were hidden, however that's not the case as it is minimized.

Any ideas welcome.

**Reply by leaanthony** [reply]
So you want this? https://stackoverflow.com/questions/814921/how-do-you-make-the-application-window-open-when-the-dock-icon-is-clicked

**Reply by keyle** [reply]
No, sorry if I wasn't being clear. 

Calling WindowHide does the expected hiding, however, alt-tabbing to the app icon should bring the window back in place visible. So should clicking the icon. Neither does.

Calling WindowMinimize cause the animation to minimise in the dock (shrinking), clicking the icon in the dock brings back the application with an animation (expanding), however alt-tabbing to it does not.

Let me know if I'm not making sense I can make a video? Either way, WindowHide is not doing a native hide I presume.

**Reply by leaanthony** [reply]
Sounds like we just need to add the hide and show roles for the menus.

**Reply by leaanthony** [reply]
This was fixed a few releases ago 👍

**Reply by keyle** [reply]
Thanks! Can you point me to the commit, tag or release?
